enhancements/olm/single-ownnamespace-enhancement-proposal.md

```diff
@@ -0,0 +1,418 @@
+   - **SingleNamespace**: `watchNamespace` differs from `spec.namespace`
+
+#### Example Configuration
+
+**OwnNamespace Mode**:
+```yaml
+apiVersion: olm.operatorframework.io/v1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  namespace: argocd                    # Install namespace
+  serviceAccount:
+    name: argocd-installer
+  config:
+    configType: Inline
+    inline:
+      watchNamespace: argocd           # Same as install namespace = OwnNamespace
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+```
+
+**SingleNamespace Mode**:
+```yaml
+apiVersion: olm.operatorframework.io/v1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  namespace: argocd                    # Install namespace
+  serviceAccount:
+    name: argocd-installer
+  config:
+    configType: Inline
+    inline:
+      watchNamespace: target-namespace # Different namespace = SingleNamespace
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+```
+
+### API Extensions
+
+#### ClusterExtension API Enhancement
+
+The ClusterExtension API is being enhanced with a `config` field designed for this purpose:
+
+```go
+type ClusterExtensionConfig struct {
+    ConfigType ClusterExtensionConfigType `json:"configType"`
+    Inline     *apiextensionsv1.JSON      `json:"inline,omitempty"`
+}
+```
+
+The `watchNamespace` configuration is specified within the `inline` JSON configuration:
+
+```json
+{
+  "watchNamespace": "target-namespace"
+}
+```
+
+#### Feature Gate
+
+The feature is controlled by the `SingleOwnNamespaceInstallSupport` feature gate:
+- **Alpha** status (default: disabled)
+- Must be explicitly enabled via `--feature-gates=SingleOwnNamespaceInstallSupport=true`
+
+### Topology Considerations
+
+#### Hypershift
+
+OLMv1 does not yet support Hypershift. Although no aspects of this feature's implementation stands out as at odds with the topology of Hypershift, it should be reviewed when OLMv1 is ready to be supported in Hypershift clusters.
+
+#### Standalone Clusters
+- **Full Compatibility**: Complete support for Single/OwnNamespace modes
+- **Standard RBAC**: Normal Role/RoleBinding generation for namespace-scoped permissions
+
+#### Single-node Deployments
+- **Resource Efficiency**: Namespace-scoped operators can reduce resource overhead on constrained single-node deployments
+- **Isolation Benefits**: Provides additional isolation even in single-node environments
+
+#### MicroShift
+
+OLMv1 does not yet support Microshift. Although no aspects of this feature's implementation stands out as odds with the topology of Microshift, it should be reviewed when OLMv1 is ready to be supported in Microshift clusters. 
+
+### Implementation Details/Notes/Constraints
+
+The ClusterExtension CRD used in the registry+v1 bundle installation process is enhanced to:
+- Parse watchNamespace configuration from the inline JSON configuration
```

Comment:
- Validate watchNamespace against the bundle's supported install modes (maybe we could add a table here to show how validation works for each install mode support configuration)